migration/src/main/scala/akka/persistence/r2dbc/migration/SqlServerMigrationToolDao.scala

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (C) 2022 - 2023 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.r2dbc.migration
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+import io.r2dbc.spi.ConnectionFactory
+import io.r2dbc.spi.Statement
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+import akka.actor.typed.ActorSystem
+import akka.annotation.InternalApi
+import akka.persistence.r2dbc.internal.Sql.Interpolation
+import akka.persistence.r2dbc.internal.codec.QueryAdapter
+import akka.persistence.r2dbc.internal.codec.SqlServerQueryAdapter
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[r2dbc] object SqlServerMigrationToolDao {
+  private val log = LoggerFactory.getLogger(classOf[SqlServerMigrationToolDao])
+
+}
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[r2dbc] class SqlServerMigrationToolDao(
+    connectionFactory: ConnectionFactory,
+    logDbCallsExceeding: FiniteDuration,
+    closeCallsExceeding: Option[FiniteDuration])(implicit ec: ExecutionContext, system: ActorSystem[_])
+    extends MigrationToolDao(connectionFactory, logDbCallsExceeding, closeCallsExceeding) {
+
+  override protected def log: Logger = SqlServerMigrationToolDao.log
+
+  override protected implicit def queryAdapter: QueryAdapter = SqlServerQueryAdapter
```

Comment:
is this used? you have imported `Sql.Interpolation`, which is without the adapter. Maybe best that we use `Sql.InterpolationWithAdapter` everywhere in this project?